src/gui/AudioInterfaceMode.h

```diff
@@ -0,0 +1,83 @@
+//*****************************************************************
+/*
+  JackTrip: A System for High-Quality Audio Network Performance
+  over the Internet
+
+  Copyright (c) 2008-2022 Juan-Pablo Caceres, Chris Chafe.
+  SoundWIRE group at CCRMA, Stanford University.
+
+  Permission is hereby granted, free of charge, to any person
+  obtaining a copy of this software and associated documentation
+  files (the "Software"), to deal in the Software without
+  restriction, including without limitation the rights to use,
+  copy, modify, merge, publish, distribute, sublicense, and/or sell
+  copies of the Software, and to permit persons to whom the
+  Software is furnished to do so, subject to the following
+  conditions:
+
+  The above copyright notice and this permission notice shall be
+  included in all copies or substantial portions of the Software.
+
+  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+  EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+  OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+  NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+  HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+  WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+  OTHER DEALINGS IN THE SOFTWARE.
+*/
+//*****************************************************************
+
+/**
+ * \file AudioInterfaceMode.h
+ * \author Matt Horton
+ * \date December 2022
+ */
+
+enum class AudioInterfaceMode {
+    JACK,     ///< Jack Mode
+    RTAUDIO,  ///< RtAudio Mode
+    BOTH,
+    NONE
+};
+
+#ifdef RT_AUDIO
+#ifndef NO_JACK
+constexpr AudioInterfaceMode mode = AudioInterfaceMode::BOTH;
```

Comment:
I know that there are (two?) places where we use have_libjack() for the weakjack case. Maybe it would be good to put that as a runtime check here. AFAIK that wouldn't contradict constexpr. But I don't speak constexpr fluently.

Reply:
I dooon't quite understand that idea, but I did run into some cases where I still needed preprocessor directives as things were not building any more.

Reply:
When using weakjack we don't know the availability of jack at compile time. But it would make sense to have this checked with isBackendAvailable(). AFAIU constexpr evaluates expressions at compile time as long as it works. So for weakjack we would introduce a runtime check for jack but everything else would be evaluated at compile time nonetheless.